components/ws-manager-mk2/controllers/workspace_controller.go

```diff
@@ -472,6 +465,19 @@ func (r *WorkspaceReconciler) deleteSecret(ctx context.Context, name, namespace
 	return err
 }
 
+// errorLogConflict logs the error if it's a conflict, instead of returning it as a reconciler error.
+// This is to reduce noise in our error logging, as conflicts are to be expected.
+// For conflicts, instead a result with `Requeue: true` is returned, which has the same requeuing
+// behaviour as returning an error.
+func errorResultLogConflict(log logr.Logger, err error) (ctrl.Result, error) {
+	if errors.IsConflict(err) {
+		log.Info("conflict error", "error", err.Error())
+		return ctrl.Result{Requeue: true}, nil
```

Comment:
This should use a delay, right? (otherwise, it will be processed by another worker immediately)

Reply:
yeah good point, looking at the controller [logs](https://cloudlogging.app.goo.gl/ogQHmm82D8huHDdd9) it does get retried pretty much immediately several times. I'll add a small delay